Number_types/include/CGAL/boost_mp.h

```diff
@@ -139,6 +141,259 @@ struct Algebraic_structure_traits<boost::multiprecision::detail::expression<T1,T
 
 // Real_embeddable_traits
 
+namespace Boost_MP_internal {
+
+  // here we know that `intv` contains int64 numbers such that their msb is std::numeric_limits<double>::digits-1
+  inline
+  Interval_nt<false> shift_positive_interval( const Interval_nt<false>& intv, const int e ) {
+    CGAL_assertion(intv.inf() > 0.0);
+    CGAL_assertion(intv.sup() > 0.0);
+    typedef std::numeric_limits<double> limits;
+
+    if (e < std::numeric_limits<double>::min_exponent)
+    {
+      if ( e+std::numeric_limits<double>::digits < std::numeric_limits<double>::min_exponent)
+        return CGAL::Interval_nt<false>(0, (limits::min)());
```

Comment:
I thought you had an assertion somewhere that intervals are tight. Is that assertion gone?

Reply:
It is in `are_bounds_correct()` and it is tight only if we did not go into the special cases around zero or at infinity.